src/wfs/src/test/java/org/geoserver/wfs/v2_0/GetFeatureTest.java

```diff
@@ -157,6 +157,33 @@ public void testGetWithCount() throws Exception {
         assertEquals("15", dom.getDocumentElement().getAttribute("numberMatched"));
     }
 
+    @Test
+    public void testGetWithCountAndStartIndex0() throws Exception {
+        Document dom = getAsDOM(
+            "wfs?request=GetFeature&typenames=cdf:Fifteen&version=2.0.0&service=wfs&count=5&startIndex=0");
+        XMLAssert.assertXpathEvaluatesTo("5", "count(//cdf:Fifteen)", dom);
+        assertEquals("5", dom.getDocumentElement().getAttribute("numberReturned"));
+        assertEquals("15", dom.getDocumentElement().getAttribute("numberMatched"));
+    }
+
+    @Test
+    public void testGetWithCountAndStartIndexMiddle() throws Exception {
+        Document dom = getAsDOM(
+            "wfs?request=GetFeature&typenames=cdf:Fifteen&version=2.0.0&service=wfs&count=5&startIndex=7");
+        XMLAssert.assertXpathEvaluatesTo("5", "count(//cdf:Fifteen)", dom);
+        assertEquals("5", dom.getDocumentElement().getAttribute("numberReturned"));
+        assertEquals("15", dom.getDocumentElement().getAttribute("numberMatched"));
+    }
+
+    @Test
+    public void testGetWithCountAndStartIndexEnd() throws Exception {
```

Comment:
But this one does :-) 
The extra test coverage of the others does not hurt, either.